src/System.Private.ServiceModel/src/System/ServiceModel/Channels/CommunicationObject.cs

```diff
@@ -312,6 +314,18 @@ protected void Fault()
             OnFaulted();
         }
 
+        internal void Fault(Exception exception)
+        {
+            lock (this.ThisLock)
```

Comment:
Yuck -- this file already has a ton of this. usage.  I'll fix the code I added but want to prevent sweeping code changes outside of the scope of this PR.

---
In reply to: [78150268](https://github.com/dotnet/wcf/pull/1515#discussion_r78150268) [](ancestors = 78150268)